src/tests/ftest/launch.py

```diff
@@ -473,7 +583,7 @@ def rename_logs(avocado_logs_dir, test_file):
         avocado_logs_dir (str): avocado job-results directory
         test_file (str): the test python file
     """
-    test_name = os.path.splitext(os.path.basename(test_file))[0]
+    test_name = get_test_category(test_file)
```

Comment:
Only because you are making this modification here to the test filename...

I have been wondering if we want to drop the _job-2019-07-26T21.09-3e19156-_ part of the _test_name_ so that the resulting name is just _container-delete_.

This is useful because when you are in the dir for the test (i.e. it would be https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-746/11/artifact/Functional/job-results/container-delete/) the _Next Build_ and _Previous Build_ navigation features: ![image](https://user-images.githubusercontent.com/559317/62941001-ecff0b80-bda2-11e9-9b4c-334a2975936c.png) would work.

They don't currently because obviously every test run has a different _date-time-hash_ stamp.

Maybe for this PR or maybe not.  But I also wonder if this change in this PR is relevant or if it should go into it's own PR, perhaps with the above change.

Reply:
What about adding this argument?
```
    parser.add_argument(
        "-j", "--jenkins",
        action="store_true",
        help="replace the date-time-hash in the avocado test logs directory "
             "with the test name for use with running in CI with Jenkins")
```
It would override the `--rename` argument.

Reply:
Hrm.  I can see the point.  While this might be nice for Jenkins, it might not be so nice outside of Jenkins where the user might run tests multiple times and want to keep the history of results.

I actually have another patch in my TODO pile here to do something similar:

```
diff --git a/Jenkinsfile b/Jenkinsfile
index d6fb3b5c..2c7bbda4 100644
--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -984,7 +984,12 @@ pipeline {
                                                test_tag=regression,vm
                                            fi
                                            tnodes=$(echo $NODELIST | cut -d ',' -f 1-9)
-                                           ./ftest.sh "$test_tag" $tnodes''',
+                                           rm -rf src/tests/ftest/avocado ./*_results.xml
+                                           mkdir -p src/tests/ftest/avocado/job-results
+                                           ./ftest.sh "$test_tag" $tnodes
+                                           # Remove the latest avocado symlink directory to avoid inclusion in the
+                                           # jenkins build artifacts
+                                           unlink src/tests/ftest/avocado/job-results/latest''',
                                 junit_files: "src/tests/ftest/avocado/*/*/*.xml src/tests/ftest/*_results.xml",
                                 failure_artifacts: env.STAGE_NAME
                     }
@@ -1051,7 +1056,12 @@ pipeline {
                                                test_tag=pr,hw
                                            fi
                                            tnodes=$(echo $NODELIST | cut -d ',' -f 1-9)
-                                           ./ftest.sh "$test_tag" $tnodes''',
+                                           rm -rf src/tests/ftest/avocado ./*_results.xml
+                                           mkdir -p src/tests/ftest/avocado/job-results
+                                           ./ftest.sh "$test_tag" $tnodes
+                                           # Remove the latest avocado symlink directory to avoid inclusion in the
+                                           # jenkins build artifacts
+                                           unlink src/tests/ftest/avocado/job-results/latest''',
                                 junit_files: "src/tests/ftest/avocado/*/*/*.xml src/tests/ftest/*_results.xml",
                                 failure_artifacts: env.STAGE_NAME
                     }
diff --git a/ftest.sh b/ftest.sh
index 9e351f0e..0c1d9fb6 100755
--- a/ftest.sh
+++ b/ftest.sh
@@ -129,11 +129,6 @@ sed -i.dist -e "s/- boro-A/- ${nodes[1]}/g" \
             -e "s/- boro-G/- ${nodes[7]}/g" \
             -e "s/- boro-H/- ${nodes[8]}/g" "${yaml_files[@]}"
 
-
-# let's output to a dir in the tree
-rm -rf src/tests/ftest/avocado ./*_results.xml
-mkdir -p src/tests/ftest/avocado/job-results
-
 trap 'set +e; cleanup' EXIT
 
 DAOS_BASE=${SL_PREFIX%/install}
@@ -315,10 +310,6 @@ else
     rc=0
 fi
 
-# Remove the latest avocado symlink directory to avoid inclusion in the
-# jenkins build artifacts
-unlink $DAOS_BASE/src/tests/ftest/avocado/job-results/latest
-
 # get stacktraces for the core files
 if ls core.*; then
     # this really should be a debuginfo-install command but our systems lag
```

which as you can see, moves functionality (cleaning up the results dir to ensure only a single result exists)  that really should only apply in the Jenkins use model to Jenkins.

I'm thinking this stripping of the date-time-hash stamp probably also belongs in the Jenkinsfile also.  So maybe I will just add it to my patch above.

Reply:
I would agree with that.